bpf/bpf_host.c

```diff
@@ -1175,8 +1175,11 @@ do_netdev(struct __ctx_buff *ctx, __u16 proto, const bool from_host)
 #ifdef ENABLE_WIREGUARD
 		if (!from_host) {
 			next_proto = ip6->nexthdr;
-			l4_off = ETH_HLEN + ipv6_hdrlen(ctx, &next_proto);
-			if (ctx_is_wireguard(ctx, l4_off, next_proto, ipcache_srcid))
+			hdrlen = ipv6_hdrlen(ctx, &next_proto);
+			if (hdrlen < 0)
+				return send_drop_notify_error(ctx, identity, hdrlen,
+						      CTX_ACT_DROP, METRIC_INGRESS);
```

Comment:
I think we shouldn't drop. It's perfectly fine for unrelated traffic to use unhandled / too many IPv6 extension headers. We can just let that traffic pass through to the host, it's certainly not Cilium WG traffic.

Reply:
Fair point, so that we avoid avoid multiples places where we potentially drop the packet.
Just adjusted it, thanks again :+1: